File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.cpp
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> intervals) {
    vector<int> answer;
    
    int a1 = intervals[0][0];
    int b1 = intervals[0][1];
    int a2 = intervals[1][0];
    int b2 = intervals[1][1];
    
    // 첫 번째 구간 (a1 ~ b1 포함)
    for (int i = a1; i <= b1; i++) {
        answer.push_back(arr[i]);
    }
    
    // 두 번째 구간 (a2 ~ b2 포함)
    for (int i = a2; i <= b2; i++) {
        answer.push_back(arr[i]);
    }
    
    return answer;
}

File: 프로그래머스/0/181895. 배열 만들기 3/배열 만들기 3.py
def solution(arr, intervals):
    a1, b1 = intervals[0]
    a2, b2 = intervals[1]
    
    # a1부터 b1까지 포함한 구간 + a2부터 b2까지 포함한 구간
    return arr[a1 : b1 + 1] + arr[a2 : b2 + 1]